### PR TITLE
Cache schemas

### LINKED ISSUE #2648: Cache schema and Avro schema in BigQueryType macro

### PR DESCRIPTION
**Pull request by jto**
fixes #2648

### CODE REVIEW

**Comment by jto** [comment]
@regadas let me know what you think

File: scio-bigquery/src/main/scala/com/spotify/scio/bigquery/types/TypeProvider.scala [hunk 1]
```diff
@@ -126,7 +149,10 @@ private[types] object TypeProvider {
     }
     val (queryFormat: String, _) :: queryArgs = extractedArgs
     val query = BigQueryPartitionUtil.latestQuery(bigquery, formatString(extractedArgs.map(_._1)))
-    val schema = bigquery.query.schema(query)
```
Comment (regadas) [inline scio-bigquery/src/main/scala/com/spotify/scio/bigquery/types/TypeProvider.scala]: Same thing here

File: scio-bigquery/src/main/scala/com/spotify/scio/bigquery/types/TypeProvider.scala [hunk 2]
```diff
@@ -86,13 +89,33 @@ private[types] object TypeProvider {
...
+      s
+    }
+  }
+
   def storageImpl(c: blackbox.Context)(annottees: c.Expr[Any]*): c.Expr[Any] = {
     import c.universe._
 
     val (table, args, selectedFields, rowRestriction) = extractStorageArgs(c)
     val tableSpec = BigQueryPartitionUtil.latestTable(bigquery, formatString(table :: args))
-    val avroSchema = bigquery.tables.storageReadSchema(tableSpec, selectedFields, rowRestriction)
```
Comment (regadas) [inline scio-bigquery/src/main/scala/com/spotify/scio/bigquery/types/TypeProvider.scala]: in this case `storageReadSchema` already caches the schema on disk.

**Comment by jto** [comment]
Closing. I don't think it really makes any difference.

**Comment by jto** [comment]
@regadas feel free to do it. I waonder if it translate to gain in a "real" compilation though.

**Pull request closed without merging**